addons/spreadsheet_dashboard/static/src/bundle/dashboard_action/dashboard_action.js

```diff
@@ -64,6 +65,25 @@ export class SpreadsheetDashboardAction extends Component {
                 return [dashboard && dashboard.model, dashboard && dashboard.status];
             }
         );
+        useEffect(() => {
+            const printLog = (event) => {
+                const dashboard = this.state.activeDashboard;
+                if (!dashboard || dashboard.status !== Status.Loaded) {
+                    return;
+                }
+                const datasources = dashboard.model.getters
+                    .getLoadedDatasources()
+                    .map((ds) => ds.source);
+                if (datasources.length) {
+                    this.rpc("/spreadsheet/log", {
+                        type: "print",
```

Comment:
instead of repeating this piece of code 3 times, you could introduce a new command like `"LOG_DATASOURCE_EXPORT"` or something like that